pkg/translator/prometheusremotewrite/number_data_points_v2_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package prometheusremotewrite
+
+import (
+	writev2 "github.com/prometheus/prometheus/prompb/io/prometheus/write/v2"
+	"testing"
+	"time"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func TestPrometheusConverterV2_addGaugeNumberDataPoints(t *testing.T) {
+	// TODO add test case with duplicate metrics
+	ts := uint64(time.Now().UnixNano())
+	tests := []struct {
+		name   string
+		metric func() pmetric.Metric
+		want   func() map[uint64]*writev2.TimeSeries
+	}{
+		{
+			name: "gauge",
```

Comment:
please add a case for `pt.Flags().NoRecordedValue()`, and for a float gauge.

Reply:
for the second one do you mean DoubleGauge as here https://github.com/jmichalek132/opentelemetry-collector-contrib/blob/11396751f11905fe62df105f444cef9ab531def0/pkg/translator/prometheusremotewrite/testutils_test.go#L230  ?

Reply:
yes, double in OTel means float64 in Go.

Reply:
done